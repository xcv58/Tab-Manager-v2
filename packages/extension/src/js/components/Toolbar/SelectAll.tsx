import React from 'react'
import { observer } from 'mobx-react-lite'
import Checkbox from '@mui/material/Checkbox'
import Tooltip from '@mui/material/Tooltip'
import { TOOLTIP_DELAY } from 'libs'
import { useStore } from 'components/hooks/useStore'

export default observer(() => {
  const { searchStore } = useStore()
  const { allTabSelected, someTabSelected, matchedTabs } = searchStore
  const title = (allTabSelected ? 'Unselect' : 'Select') + ' all tabs'
  return (
    <Tooltip title={title} enterDelay={TOOLTIP_DELAY}>
      <div className="flex">
        <Checkbox
          color="primary"
          checked={allTabSelected}
          disabled={matchedTabs.length === 0}
          onChange={(e) => {
            e.target.blur()
            searchStore.toggleSelectAll()
          }}
          indeterminate={someTabSelected}
          inputProps={{ 'aria-label': title }}
        />
      </div>
    </Tooltip>
  )
})
