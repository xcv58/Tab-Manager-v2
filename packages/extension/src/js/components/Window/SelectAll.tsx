import React from 'react'
import { observer } from 'mobx-react-lite'
import Checkbox from '@mui/material/Checkbox'
import Tooltip from '@mui/material/Tooltip'

export default observer(({ win }) => {
  const { allTabSelected, someTabSelected, disableSelectAll, toggleSelectAll } =
    win
  const title = `${allTabSelected ? 'Unselect' : 'Select'} all tabs`
  return (
    <Tooltip title={title}>
      <Checkbox
        color="primary"
        inputProps={{
          'aria-label': title,
        }}
        disabled={disableSelectAll}
        checked={allTabSelected}
        onChange={(e) => {
          e.target.blur()
          toggleSelectAll()
        }}
        indeterminate={someTabSelected || disableSelectAll}
      />
    </Tooltip>
  )
})
