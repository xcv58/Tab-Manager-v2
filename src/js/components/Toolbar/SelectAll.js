import React from 'react'
import { inject, observer } from 'mobx-react'
import Checkbox from 'material-ui/Checkbox'
import Tooltip from 'material-ui/Tooltip'

@inject('searchStore')
@observer
export default class SelectAll extends React.Component {
  selectAll = (e) => {
    e.target.blur()
    const {
      allTabSelected, selectAll, unselectAll
    } = this.props.searchStore
    if (allTabSelected) {
      unselectAll()
    } else {
      selectAll()
    }
  }

  render () {
    const { allTabSelected, someTabSelected } = this.props.searchStore
    const title = (allTabSelected ? 'Unselect' : 'Select') + ' all tabs'
    return (
      <Tooltip title={title}>
        <div>
          <Checkbox
            checked={allTabSelected}
            onChange={this.selectAll}
            indeterminate={someTabSelected}
          />
        </div>
      </Tooltip>
    )
  }
}
