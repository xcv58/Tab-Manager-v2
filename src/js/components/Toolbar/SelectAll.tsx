import React from 'react'
import { observer } from 'mobx-react'
import Checkbox from '@material-ui/core/Checkbox'
import Tooltip from '@material-ui/core/Tooltip'
import { TOOLTIP_DELAY } from 'libs'
import { useStore } from 'components/StoreContext'

export default observer(() => {
  const { searchStore } = useStore()
  const { allTabSelected, someTabSelected } = searchStore
  const title = (allTabSelected ? 'Unselect' : 'Select') + ' all tabs'
  return (
    <Tooltip title={title} enterDelay={TOOLTIP_DELAY}>
      <div>
        <Checkbox
          color='primary'
          checked={allTabSelected}
          onChange={e => {
            e.target.blur()
            const { selectAll, unselectAll } = searchStore
            if (allTabSelected) {
              unselectAll()
            } else {
              selectAll()
            }
          }}
          indeterminate={someTabSelected}
          inputProps={{ 'aria-label': title }}
        />
      </div>
    </Tooltip>
  )
})
