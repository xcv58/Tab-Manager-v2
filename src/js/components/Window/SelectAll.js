import React from 'react'
import { inject, observer } from 'mobx-react'
import Checkbox from 'material-ui/Checkbox'
import Tooltip from 'material-ui/Tooltip'

@inject('tabStore')
@observer
export default class SelectAll extends React.Component {
  selectAll = e => {
    e.target.blur()
    const {
      win: { allTabSelected, matchedTabs },
      tabStore: { selectAll, unselectAll }
    } = this.props
    if (allTabSelected) {
      unselectAll(matchedTabs)
    } else {
      selectAll(matchedTabs)
    }
  }

  render () {
    const { allTabSelected, someTabSelected, disableSelectAll } = this.props.win
    const title = `${allTabSelected ? 'Unselect' : 'Select'} all tabs`
    return (
      <Tooltip title={title}>
        <Checkbox
          color='primary'
          disabled={disableSelectAll}
          checked={allTabSelected}
          onChange={this.selectAll}
          indeterminate={someTabSelected || disableSelectAll}
        />
      </Tooltip>
    )
  }
}
