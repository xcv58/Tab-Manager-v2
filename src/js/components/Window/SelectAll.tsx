import React from 'react'
import { observer } from 'mobx-react'
import Checkbox from '@material-ui/core/Checkbox'
import Tooltip from '@material-ui/core/Tooltip'
import { useStore } from 'components/StoreContext'

export default observer(({ win }) => {
  const { tabStore } = useStore()
  const { allTabSelected, someTabSelected, disableSelectAll } = win
  const title = `${allTabSelected ? 'Unselect' : 'Select'} all tabs`
  return (
    <Tooltip title={title}>
      <Checkbox
        color='primary'
        disabled={disableSelectAll}
        checked={allTabSelected}
        onChange={e => {
          e.target.blur()
          const { allTabSelected, matchedTabs } = win
          const { selectAll, unselectAll } = tabStore
          if (allTabSelected) {
            unselectAll(matchedTabs)
          } else {
            selectAll(matchedTabs)
          }
        }}
        indeterminate={someTabSelected || disableSelectAll}
      />
    </Tooltip>
  )
})
