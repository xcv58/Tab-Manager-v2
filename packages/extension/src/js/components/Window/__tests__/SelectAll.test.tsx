import React from 'react'
import { shallow } from 'enzyme'
import Tooltip from '@material-ui/core/Tooltip'
import Checkbox from '@material-ui/core/Checkbox'
import SelectAll from 'components/Window/SelectAll'

const id = 'id'
const toggleSelectAll = jest.fn()
const props = {
  win: {
    id,
    toggleSelectAll,
    allTabSelected: true,
    someTabSelected: false,
    disableSelectAll: false
  }
}

describe('SelectAll', () => {
  it('should render correct components', () => {
    const el = shallow(<SelectAll {...props} />)
    expect(el.find(Tooltip).length).toBe(1)
    expect(el.find(Checkbox).length).toBe(1)
  })

  it('should render Tooltip title based on allTabSelected', () => {
    let el = shallow(<SelectAll {...props} />)
    expect(el.find(Tooltip).length).toBe(1)
    expect(el.find(Tooltip).props().title).toBe('Unselect all tabs')
    el = shallow(<SelectAll {...props} win={{ allTabSelected: false }} />)
    expect(el.find(Tooltip).length).toBe(1)
    expect(el.find(Tooltip).props().title).toBe('Select all tabs')
  })

  it('should call toggleSelectAll when click', () => {
    const el = shallow(<SelectAll {...props} />)
    const blur = jest.fn()
    el.find(Checkbox).props().onChange({ target: { blur } })
    expect(blur.mock.calls.length).toBe(1)
    expect(toggleSelectAll.mock.calls.length).toBe(1)
  })
})
