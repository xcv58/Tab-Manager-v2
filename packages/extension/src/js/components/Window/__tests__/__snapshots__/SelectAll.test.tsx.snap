// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectAll should render Tooltip title based on allTabSelected 1`] = `
<div
  class="MuiTooltip-popper MuiTooltip-popperInteractive css-yem0l2-MuiPopper-root-MuiTooltip-popper MuiPopperUnstyled-root"
  data-popper-escaped=""
  data-popper-placement="bottom"
  data-popper-reference-hidden=""
  id=":r1:"
  role="tooltip"
  style="position: absolute; top: 0px; left: 0px; margin: 0px; transform: translate(0px, 0px);"
>
  <div
    class="MuiTooltip-tooltip MuiTooltip-tooltipPlacementBottom css-ja5taz-MuiTooltip-tooltip"
    style="opacity: 1; transform: scale(1, 1); transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 133ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    Unselect all tabs
  </div>
</div>
`;

exports[`SelectAll should render correct components 1`] = `
<div>
  <span
    aria-label="Unselect all tabs"
    class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
    data-mui-internal-clone-element="true"
  >
    <input
      aria-label="Unselect all tabs"
      checked=""
      class="PrivateSwitchBase-input css-1m9pwf3"
      data-indeterminate="false"
      type="checkbox"
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="CheckBoxIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
      />
    </svg>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </span>
</div>
`;
